app.py

```diff
@@ -0,0 +1,437 @@
+    :param payment_status: info from citybase payload
+    :return: json object to insert in knack citybase_messages table
+    """
+    return json.dumps(
+        {
+            field_maps[environment]["messages_invoice_id"]: knack_invoice,
+            field_maps[environment]["messages_connected_invoice"]: knack_invoice,
+            field_maps[environment]["messages_created_date"]: today_date,
+            field_maps[environment]["messages_status"]: payment_status,
+            field_maps[environment]["messages_citybase_id"]: citybase_id,
+        }
+    )
+
+
+def update_parent_reservation(environment, knack_record_id, today_date):
+    """
+    Uses a knack_record_id to look up the transaction in knack, then finds the appropriate connection field
+    And sends payload to knack, marking parent reservation payment received status as TRUE
+    """
+    record_response = requests.get(
+        f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID}/records/{knack_record_id}",
+        headers=headers[environment],
+    )
+    record_data = record_response.json()
+    if record_data[field_maps[environment]["ots_connection_field"]]:
+        # get parent record id
+        parent_record_response = record_data[
+            field_maps[environment]["ots_connection_field"] + "_raw"
+        ]
+        parent_record_id = parent_record_response[0]["id"]
+        ots_payload = json.dumps(
+            {
+                field_maps[environment]["ots_application_status"]: "Approved",
+                field_maps[environment]["ots_paid_status"]: True,
+                field_maps[environment]["ots_payment_date"]: today_date,
+            }
+        )
+        parent_update_response = requests.put(
+            f"{KNACK_API_URL}{OTS_OBJECT_ID}/records/{parent_record_id}",
+            headers=headers[environment],
+            data=ots_payload,
+        )
+        print(parent_update_response)
+    if record_data[field_maps[environment]["lpb_connection_field"]]:
+        # get parent record id
+        parent_record_response = record_data[
+            field_maps[environment]["lpb_connection_field"] + "_raw"
+        ]
+        parent_record_id = parent_record_response[0]["id"]
+        lpb_payload = json.dumps(
+            {
+                field_maps[environment]["lpb_application_status"]: "Approved",
+                field_maps[environment]["lpb_paid_status"]: True,
+                field_maps[environment]["lpb_payment_date"]: today_date,
+            }
+        )
+        parent_update_response = requests.put(
+            f"{KNACK_API_URL}{LPB_OBJECT_ID}/records/{parent_record_id}",
+            headers=headers[environment],
+            data=lpb_payload,
+        )
+        print(parent_update_response)
+
+
+@app.route("/")
+def index():
+    now = datetime.now().isoformat()
+    return f"Austin Transportation Public Works Department Citybase healthcheck {now}"
+
+
+@app.route("/citybase_postback", methods=["POST"])
```

Comment:
This is a sign of my inexperience with Flask apps, because I didn't know this was an option! and thought I had to define the routes explicitly. 

I cant decide if I should try to refactor this now, or make a backlog issue for it.

Reply:
Since this is in production, I think either way is viable, because I can't imagine the UAT route is going to get called any. I'd almost be inclined to put a `return("UAT has concluded for this product.")` statement right at the top of that subroutine in case it does or to remove the UAT version entirely. 

I think if you wanted to backlog an issue to refactor it, I think that would be fine, but I think that that refactor should happen before any future work that might come down the line so that you or whomever does that work doesn't have to handle keeping two functions in sync manually. That doesn't mean it needs to get in this PR at all though. Maybe a comment pointing to the backlog issue though so future developers find that bread crumb.

Reply:
I'm not loving that the app does double-duty for UAT and prod. i'd much rather see the redundant code slashed and a separate test instance deployed. are there some external constraints around that approach? e.g. citybase can't point to a separate test instance?

Reply:
I am really liking this idea of nixing the UAT endpoint now that Citybase no longer needs to test. They required a UAT endpoint, but now I think if we were to need to test anything locally, we would just use a different environment file. 

I've been trying to think about what other needs we would have for the UAT endpoint. I think I'll go with replying with the UAT message Frank suggested, just in case citybase tries to interact with it.